src/plugins/gs-plugin-eos.c

```diff
@@ -514,6 +533,25 @@ app_is_proxy (GsApp *app)
 }
 
 static gboolean
+app_is_compatible_with_os (GsPlugin *plugin, GsApp *app)
+{
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	const char *app_available_since;
+
+	if (!priv->os_version_id)
+		return TRUE;
+
+	app_available_since =
+		gs_app_get_metadata_item (app, "EndlessOS::available-since");
+	if (!app_available_since)
+		return TRUE;
+
+	/* if the OS version is greater than the app "available-since"
+	 * metadata item, it means it is incompatible */
+	return as_utils_vercmp (priv->os_version_id, app_available_since) > 0;
```

Comment:
`as_utils_vercmp()` is documented as returning positive values when the first argument is a greater version than the second.
So the code gets the check right (if OS version is e.g. 3.1 it means it's compatible with an app available since 3.0.7), but the comment is wrong. Additionally, if both of those are the same version (function returning zero) it should still say the app is compatible. So the check should be `>= 0` I think